newdle/api.py

```diff
@@ -307,52 +308,53 @@ def create_newdle(title, duration, timezone, timeslots, participants, private, n
         notify=notify,
     )
     db.session.add(newdle)
+    db.session.flush()
+    send_invitation_emails(newdle)
     db.session.commit()
-    notify_newdle_participants(
-        newdle,
-        f'Invitation: {newdle.title}',
-        'invitation_email.txt',
-        'invitation_email.html',
-        lambda p: {
-            'creator': newdle.creator_name,
-            'title': newdle.title,
-            'participant': p.name,
-            'summary_link': url_for('newdle_summary', code=newdle.code, _external=True),
-            'answer_link': url_for(
-                'newdle', code=newdle.code, participant_code=p.code, _external=True
-            ),
-        },
-    )
     return NewdleSchema().jsonify(newdle)
 
 
-@api.route('/newdle/<code>')
-@allow_anonymous
-def get_newdle(code):
-    newdle = Newdle.query.filter_by(code=code).first_or_404(
-        'Specified newdle does not exist'
-    )
-    if newdle.deleted:
-        return DeletedNewdleSchema().jsonify(newdle)
-    return RestrictedNewdleSchema().jsonify(newdle)
-
-
 @api.route('/newdle/<code>', methods=('PATCH',))
-@use_args(UpdateNewdleSchema(), locations=('json',))
+@use_args(NewNewdleSchema(partial=True), locations=('json',))
 def update_newdle(args, code):
     newdle = Newdle.query.filter_by(code=code).first_or_404(
         'Specified newdle does not exist'
     )
-    if newdle.creator_uid != g.user['uid']:
+    if g.user is None or newdle.creator_uid != g.user['uid']:
         raise Forbidden
+    new_participants = []
+    if 'participants' in args:
+        participants = args.pop('participants')
+        # Filter the new participants to be created (left diff)
+        uids = set(p.auth_uid for p in newdle.participants)
+        new_participants = [
```

Comment:
This is the same as the first revision, right?

Reply:
not sure what you mean - my suggestion is using a set instead of a list, so you can use `|=` below

Reply:
Ah, I confused the lines. Sounds good.
`notify_newdle_participants` in the lines below expects `participants` to be a list, but since they are handled in the same way (at least for now), I guess this is fine.

Reply:
I think it works fine with a set... at least I don't see any operation in `notify_newdle_participants` that wouldn't work with a set.